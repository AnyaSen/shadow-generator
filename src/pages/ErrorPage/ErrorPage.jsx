import React from "react";
import Styles from "./ErrorPage.module.scss";
import { Link } from "react-router-dom";

import SecondaryButton from "../../components/Buttons/SecondaryButton/SecondaryButton";
import Layout from "../../components/Layout/Layout";

export default function ErrorPage() {
  return (
    <Layout>
      <div className={Styles.errorContainer}>
        <h1 className={Styles.error}>Sorry, an error has occured!</h1>
        <Link to="/">
          <SecondaryButton text="Go Back" />
        </Link>
      </div>
    </Layout>
  );
}
