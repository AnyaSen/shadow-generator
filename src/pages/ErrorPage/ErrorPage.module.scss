@import "../../assets/scss/variables";

.errorContainer {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  h1 {
    font-size: 3rem;
  }
}
