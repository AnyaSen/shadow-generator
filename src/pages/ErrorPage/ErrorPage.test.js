import React from "react";
import { BrowserRouter } from "react-router-dom";
import { render, cleanup } from "@testing-library/react";

import ErrorPage from "./ErrorPage";

describe("<ErrorPage />", () => {
  afterEach(cleanup);

  test("Should have correct text and have 2 children.", () => {
    const { getByText, getByTestId, container } = render(
      <BrowserRouter>
        <ErrorPage />
      </BrowserRouter>
    );

    const heading = getByText("Sorry, an error has occured!");

    expect(heading).toBeTruthy();

    const ErrorPageContainer = getByTestId("Error");

    expect(ErrorPageContainer.children.length).toEqual(2);

    expect(container.firstChild).toMatchSnapshot();
  });
});
