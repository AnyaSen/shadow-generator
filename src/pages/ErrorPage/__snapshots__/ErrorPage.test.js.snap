// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorPage /> Should have correct text and have 2 children. 1`] = `
<div
  class="Layout"
  data-testid="Layout"
>
  <div
    class="Header"
  >
    <a
      href="/"
    >
      <h1>
        SHADOW GENERATOR
      </h1>
    </a>
  </div>
  <div
    class="childrenContainer"
  >
    <div
      class="errorContainer"
      data-testid="Error"
    >
      <h1
        class="error"
      >
        Sorry, an error has occured!
      </h1>
      <a
        href="/"
      >
        <button
          class="Button"
        >
          Go Back
        </button>
      </a>
    </div>
  </div>
  <div
    class="Footer"
  >
    <p>
      Created by
       
      <a
        href="https://portfolio-anna-senchikhina.netlify.com/"
        rel="noopener noreferrer"
        target="_blank"
      >
        Anna Senchikhina
      </a>
    </p>
  </div>
</div>
`;
